clients/ui/bff/internal/api/middleware.go

```diff
@@ -66,14 +67,23 @@ func (app *App) InjectUserHeaders(next http.Handler) http.Handler {
 	})
 }
 
-func (app *App) enableCORS(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// TODO(ederign) restrict CORS to a much smaller set of trusted origins.
-		// TODO(ederign) deal with preflight requests
-		w.Header().Set("Access-Control-Allow-Origin", "*")
+func (app *App) EnableCORS(next http.Handler) http.Handler {
+	allowedOrigins, ok := ParseOriginList(app.config.AllowedOrigins)
```

Comment:
It's a nitpick and can be done in a FUP PR. 

As we are not planning to change AllowedOrigins values in runtime, we can parse them once during initialization and avoid dois this call on every request.

Reply:
Yeah that's a really good point actually - I'll put a follow up in today just to fix these couple